File: src/main/java/ru/translater/collect/CollectProperties.java
package ru.translater.collect;

import lombok.Data;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import ru.translater.common.CommonProperties;

import java.util.List;

@Data
@Configuration
@ConfigurationProperties(prefix = "collect")
public class CollectProperties extends CommonProperties {
    private Integer numberOfCrawlers;
    private List<String> seedUrls;
    private List<String> filterUrls;
    private List<String> notFilterUrls;
}

File: src/main/kotlin/com/translater/collect/properties/CollectProperties.kt
package com.translater.collect.properties

import com.translater.common.properties.CommonProperties
import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding

@ConstructorBinding
@ConfigurationProperties(prefix = "collect")
data class CollectProperties(
    override val storePath: String,
    val numberOfCrawlers: Int,
    val seedUrls: List<String>,
    val filterUrls: List<String>,
    val notFilterUrls: List<String>,
    val notAvailableExtensions: String
) : CommonProperties(storePath)
